### DISCUSSION TITLE
If i used class A extends class B, when refresh page, class A will lose methods in class B

### DISCUSSION THREAD

**Discussion by Leo-tee** [vuejs/pinia#2596] [Help and Questions] [answered]
i known the reason maybe is : prototype
but how can i use  if  i need class B method, but now is undefined
```js
# demo
class A {
  say() {
    console.log('say')
}
}
export default defineStore('store', {
  state: () => {
  a : new A()
}
})
# than  refresh page ,  if i do  a.say() ==> undefined
```

**Comment by posva** [comment, marked as answer]
You need to add a custom serialization mechanism for classes to work with SSR (or local storage if that's what you are using). I need to add a cookbook entry page in the docs based on [one of my lessons in Mastering Pinia](https://masteringpinia.com/lessons/complex-state-serialization) but it sums up to using payload plugins in nuxt or https://github.com/Rich-Harris/devalue to serialize/revive classes

**Reply by Leo-tee** [reply]
You are great! Thank you!
